hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/trigger/Trigger.java

```diff
@@ -28,69 +34,144 @@
  * @author Jay Shaughnessy
  * @author Lucas Ponce
  */
-public class Trigger extends TriggerTemplate {
+public class Trigger {
 
-    public enum Mode {
-        FIRING, AUTORESOLVE
-    };
+    @JsonInclude
+    private String tenantId;
 
+    /** Unique within the tenant */
     @JsonInclude
     private String id;
 
+    /** For display */
+    @JsonInclude
+    private String name;
+
+    @JsonInclude(Include.NON_EMPTY)
+    protected Map<String, String> context;
+
+    /** A map with key based on actionPlugin and value a set of action's ids */
+    @JsonInclude(Include.NON_EMPTY)
+    private Map<String, Set<String>> actions;
+
+    /** Disable automatically after firing */
+    @JsonInclude
+    private boolean autoDisable;
+
+    /** Enable automatically if disabled and resolved manually */
+    @JsonInclude
+    private boolean autoEnable;
+
+    /** Switch to auto-resolve mode after firing */
+    @JsonInclude
+    private boolean autoResolve;
+
+    /** Resolve all unresolved alerts when auto-resolve condition-set is satisfied */
+    @JsonInclude
+    private boolean autoResolveAlerts;
+
+    @JsonInclude
+    private Match autoResolveMatch;
+
+    /** Is a member trigger of the specified group triggerId */
+    @JsonInclude(Include.NON_EMPTY)
+    private String memberOf;
```

Comment:
So, a null value will identify a "regular" Trigger, a Trigger who is not a member of any Group, and it is not a Group Trigger itself.
I guess that this flag could be incompatible with Trigger.group field, right ?
Then we have three states (Group, Member, and normal that is not group nor member).